src/components/ViewRequest/DetailSections/UpdatePostId.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogBody,
+  DialogContent,
+  DialogSurface,
+  DialogTitle,
+  DialogTrigger,
+  Input,
+  Label,
+} from "@fluentui/react-components";
+import { usePostRequest } from "api/postRequestApi";
+import { useRequest } from "api/requestsApi";
+import { Controller, SubmitHandler, useForm } from "react-hook-form";
+import { useParams } from "react-router-dom";
+
+const POSTTYPES = [
+  "jobBoardPostId",
+  "joaPostId",
+  "linkedInPostId",
+  "linkedInSearchId",
+  "resumeSearchId",
+  "usaJobsPostId",
+] as const;
+
+const UpdatePostId = ({
+  detailSelection,
+}: {
+  detailSelection: (typeof POSTTYPES)[number];
+}) => {
+  const params = useParams();
+  const request = useRequest(Number(params.requestId));
+  const postRequest = usePostRequest();
+
+  const { control, handleSubmit, reset } = useForm<{
+    ItemId: string;
+  }>({
+    defaultValues: { ItemId: request.data?.[detailSelection] },
+  });
+
+  const onSubmit: SubmitHandler<{ ItemId: string }> = (data) => {
+    const requestData = {
+      ...{ [detailSelection]: data.ItemId },
+    };
+
+    postRequest.mutate({
+      requestId: Number(params.requestId),
+      postRequest: requestData,
+    });
+  };
+
+  return (
+    <Dialog
+      modalType="modal"
+      onOpenChange={() =>
+        reset({
+          ItemId: request.data?.[detailSelection],
+        })
+      }
+    >
+      <DialogTrigger disableButtonEnhancement>
+        <Button appearance="subtle" size="small" style={{ marginLeft: "auto" }}>
+          Update ID
+        </Button>
```

Comment:
This doesn't align nicely with the text for some reason -- might look better as just a pencil edit icon?